pyomo/contrib/mpc/data/series_data.py

```diff
@@ -181,6 +184,9 @@ def get_interpolated_data_at_time(self, time=None):
         is_iterable = _is_iterable(time)
         if not is_iterable:
             time = [time]
+        for t in time:
+            if t > self._time[-1] + tolerance or t < self._time[0] - tolerance:
```

Comment:
With `<` and `>` here, this will fail with a tolerance of zero (the default) if the user provides one of the endpoints. Can you update to use `>=` and `<=`?

Reply:
Unless you think the endpoints should not be valid for interpolation with a tolerance of zero. I'm open to this argument.

Reply:
You're right should have used >= and <=.  I'll fix that.

Reply:
On second look, I realized that the these are a check for whether an exception should be raised.  I think the original way was right.  When the time is equal to the first or last point and exception should not be raised.

Reply:
You're right! Can you add a test with a time point at each endpoint (and default tolerance) so we don't accidentally break this later?

Reply:
Sure! will do.